docs: add staging images to release process (#5050)
kubernetes-sigs/external-dns: docs/release.md

Currently we don't release regularly. Whenever we think it makes sense to release a new version we do it, but we aim to do a new release every month. You might want to ask in our Slack channel [external-dns](https://kubernetes.slack.com/archives/C771MKDKQ) when the next release will come out.

## Staging Release cycle

A new staging image is released weekly and can be found at [gcr.io/k8s-staging-external-dns/external-dns](https://console.cloud.google.com/gcr/images/k8s-staging-external-dns/GLOBAL/external-dns?pli=1&inv=1&invt=AboL6Q).

> > Consider the potential time lag between merging changes into the master branch and the subsequent creation of the staging image.

Reviewer: ```suggestion
> There is a time lag between merging changes into the master branch and the subsequent creation of the staging image.
```
